addons/sale_mrp/models/account_move.py

```diff
@@ -29,3 +29,26 @@ def _stock_account_get_anglo_saxon_price_unit(self):
                 price_unit = self.product_id.uom_id._compute_price(price_unit, self.product_uom_id)
         return price_unit
 
+    def _get_related_stock_moves(self, product, stock_moves):
+        product_invoice_lines = self.filtered(lambda line: line.product_id == product)
+        if not product_invoice_lines.sale_line_ids:
+            return super(AccountMoveLine, self)._get_related_stock_moves(product, stock_moves)
+
+        bom = self.env['mrp.bom']._bom_find(product=product)
+        if not bom or bom.type != 'phantom':
+            return super(AccountMoveLine, self)._get_related_stock_moves(product, stock_moves)
+
+        bom_stock_moves = self.env['stock.move']
+        for line in product_invoice_lines.sale_line_ids:
+            components = line._get_bom_component_qty(bom)
+
+            def pred(stock_move):
+                # Search for the stock moves with the product and product quantity same to the sale line's.
+                # Additionally, include only those stock moves that have account move lines that are not yet fully reconciled.
+                correct_product_and_qty = stock_move.product_id.id == product_id and stock_move.product_uom_qty == line.product_uom_qty * components[product_id]['qty']
```

Comment:
i'm not sure to understand here?

Reply:
@sle-odoo I'm sorry. I realized that my implementation here is complicated and might be even be unreliable as it relies on comparing quantities when finding moves. I think I found I better way. I'll ping you if I'm done with it.